clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -71,8 +113,26 @@ public AbstractConfig(ConfigDef definition, Map<?, ?> originals, boolean doLog)
             logAll();
     }
 
+    /**
+     * Construct a configuration with a ConfigDef, the configuration properties, optional {ConfigProviders}.
+     *
+     * @param definition the definition of the configurations; may not be null
+     * @param originals of config provider configs as well as configuration properties; may not be null
+     */
     public AbstractConfig(ConfigDef definition, Map<?, ?> originals) {
-        this(definition, originals, true);
+        this(definition, originals, Collections.emptyMap(), false);
+    }
+
+    /**
+     * Construct a configuration with a ConfigDef, the configuration properties, optional {ConfigProviders}.
+     *
+     * @param definition the definition of the configurations; may not be null
+     * @param originals of config provider configs as well as configuration properties; may not be null
```

Comment:
Nit: this is really primarily where the configuration properties go, plus any optional config provider properties. How about:
```suggestion
     * @param originals the configuration properties plus any optional config provider properties; may not be null
```